server/svix-server/src/v1/endpoints/stats.rs

```diff
@@ -0,0 +1,297 @@
+use crate::core::types::ApplicationIdOrUid;
+use crate::error::{HttpError, ValidationErrorItem};
+use crate::v1::utils::openapi_tag;
+use crate::{ctx, err_generic, err_validation, AppState};
+use aide::axum::routing::get_with;
+use aide::axum::ApiRouter;
+use axum::extract::{Path, Query, State};
+use axum::Json;
+use chrono::{DateTime, Timelike, Utc};
+use schemars::JsonSchema;
+use sea_orm::{ConnectionTrait, FromQueryResult, Statement};
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+use svix_server_derive::aide_annotate;
+
+/// Period length for a statistics data point
+#[derive(Deserialize, Serialize, JsonSchema, Debug, Eq, PartialEq, Copy, Clone)]
+#[allow(unused)]
+pub enum StatisticsPeriod {
+    OneDay,
+    FiveMinutes,
+}
+
+#[derive(Deserialize, Serialize, JsonSchema)]
+pub struct BaseStatisticsResponse {
+    pub start_date: DateTime<Utc>,
+    pub end_date: DateTime<Utc>,
+    pub period: StatisticsPeriod,
+}
+
+#[derive(Deserialize, Serialize, JsonSchema)]
+pub struct AttemptStatisticsData {
+    pub success_count: Option<Vec<i64>>,
+    pub failure_count: Option<Vec<i64>>,
+}
+
+#[derive(Deserialize, Serialize, JsonSchema)]
+pub struct AttemptStatisticsResponse {
+    #[serde(flatten)]
+    pub common_: BaseStatisticsResponse,
+    pub data: AttemptStatisticsData,
+}
+
+#[derive(Deserialize, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+#[allow(unused)]
+struct StatisticsQueryParams {
+    start_date: Option<DateTime<Utc>>,
+    end_date: Option<DateTime<Utc>>,
+}
```

Comment:
I saw a comment in the original python that mentioned "camelCase for external API, but snake_case for response." 

The tests seem to suggest this is the case, but I'm still not sure about the JSON encode/decode on the Python side to know if that was actually what was happening.

@svix-lucho if you've got callsites on the frontend expecting these responses, it'd help me to confirm if you see snake_case or camelCase field names for the response body.

Reply:
N.b. the stubs we have for this endpoint's doc generation look like they are advertising camelCased fields. I can rename the fields, I just wasn't sure if they should be or not 😰

Reply:
Looking at the inheritance tree, it looks like all fields for responses are camelCased by default so I'll go ahead and just do it. Tell me if you see different, Luco!

Reply:
I'm a bit late here, but checking the frontend it seems we expect camelCase everywhere. Just tested the branch and it seems to work just fine :)

<img width="1635" alt="Screenshot 2023-06-19 at 4 35 41 PM" src="https://github.com/svix/svix-webhooks/assets/133019767/a5440f54-7ffe-4b4f-ba66-7319aec724c0">

Reply:
(without any changes in frontend) - so the format you have here is correct

Reply:
Great. I think the comments I saw along the way must have been outdated then. All good!